plugins/BEdita/Core/tests/TestCase/Model/Action/SignupUserActionTest.php

```diff
@@ -180,13 +201,22 @@ public function testExecute($expected, array $data)
             static::assertTrue(is_string($arguments[3]));
         });
 
-        $action = new SignupUserAction();
-        $result = $action($data);
+        try {
+            $action = new SignupUserAction();
+            $result = $action($data);
+            $success = true;
+        } catch (CakeException $e) {
+            $success = $e;
+            static::assertEquals($expected->getAttributes(), $e->getAttributes());
+            static::assertEquals($expected->getCode(), $e->getCode());
+        }
```

Comment:
I'd change this test a bit:

```php
// Restore this part:
if ($expected instanceof \Exception) {
    static::expectException(get_class($expected));
}

// ...

try {
    $action = new SignupUserAction();
    $result = $action($data);
} catch (CakeException $e) {
    static::assertEquals($expected->getAttributes(), $e->getAttributes());
    static::assertEquals($expected->getCode(), $e->getCode());

    throw $e; // Re-throw exception.
}

// Leave below as it was before.
```

Basically, the test remains as it was before. We only catch the exception for the time strictly necessary to perform some additional assertions on it, and then re-throw it to continue the usual flow.

Reply:
Seems ok, applying what you suggest

Reply:
done